File: Homeworks/JS-basics/Conditional-Statements/sortThreeNumbers.js
// Sort 3 real values in descending order using nested if statements.

var numbers = [-2, 7, 3];

numbers = sortThree(numbers);
console.log(numbers);

function sortThree(arr) {
    var a = arr[0], b = arr[1], c = arr[2];

    var sorted = [];

    if(a >= b){
        if(b >= c){
            sorted.push(a, b, c);
        }
        else if(c > a){
            sorted.push(c, a, b);
        }
        else{
            sorted.push(a, c, b);
        }
    }
    else{
        if(b >= c){
            if(c >= a){
                sorted.push(b, c, a);
            }
            else{
                sorted.push(b, a, c);
            }
        }
        else{
            sorted.push(c, b, a);
        }
    }

    return sorted;
};

File: Homeworks/C#1/Conditional-Statements/SortThreeNumbers/SortThreeNumbers.cs
//Write a program that enters 3 real numbers and prints them sorted in descending order.
//Use nested if statements.
//Note: Don’t use arrays and the built-in sorting functionality.

namespace SortThreeNumbers
{
    using System;
    class SortThreeNumbers
    {
        static void Main(string[] args)
        {
            Console.Write("Enter the first number: ");
            double a = double.Parse(Console.ReadLine());
            Console.Write("Enter the second number: ");
            double b = double.Parse(Console.ReadLine());
            Console.Write("Enter the third number: ");
            double c = double.Parse(Console.ReadLine());

            //abcX
            //acbX
            //bacX
            //bcaX
            //cabX
            //cbaX

            if (a >= b)
            {
                if (b >= c) Print(a, b, c);
                else if (c >= a) Print(c, a, b);
                else Print(a, c, b);
            }
            else if (b >= a)
            {
                if (a >= c) Print(b, a, c);
                else if (c >= b) Print(c, b, a);
                else Print(b, c, a);
            }

        }

        private static void Print(double a, double b, double c)
        {
            Console.Clear();
            Console.WriteLine("{0}  {1}  {2}", a, b, c);
        }
    }
}
